examples/monit-4.2/monit.1
.\" Automatically generated by Pod::Man v1.37, Pod::Parser v1.14
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sh \" Subsection heading
.br
.if t .Sp
.ne 5
.PP
\fB\\$1\fR
.PP
..
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" Set up some character translations and predefined strings.  \*(-- will
.\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
.\" double quote, and \*(R" will give a right double quote.  | will give a
.\" real vertical bar.  \*(C+ will give a nicer C++.  Capital omega is used to
.\" do unbreakable dashes and therefore won't be available.  \*(C` and \*(C'
.\" expand to `' in nroff, nothing in troff, for use with C<>.
.tr \(*W-|\(bv\*(Tr
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.ie n \{\
.    ds -- \(*W-
.    ds PI pi
.    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
.    ds L" ""
.    ds R" ""
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds -- \|\(em\|
.    ds PI \(*p
.    ds L" ``
.    ds R" ''
'br\}
.\"
.\" If the F register is turned on, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.Sh), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.if \nF \{\
.    de IX
.    tm Index:\\$1\t\\n%\t"\\$2"
..
.    nr % 0
.    rr F
.\}
.\"
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.hy 0
.if n .na
.\"
.\" Accent mark definitions (@(#)ms.acc 1.5 88/02/08 SMI; from UCB 4.2).
.\" Fear.  Run.  Save yourself.  No user-serviceable parts.
.    \" fudge factors for nroff and troff
.if n \{\
.    ds #H 0
.    ds #V .8m
.    ds #F .3m
.    ds #[ \f1
.    ds #] \fP
.\}
.if t \{\
.    ds #H ((1u-(\\\\n(.fu%2u))*.13m)
.    ds #V .6m
.    ds #F 0
.    ds #[ \&
.    ds #] \&
.\}
.    \" simple accents for nroff and troff
.if n \{\
.    ds ' \&
.    ds ` \&
.    ds ^ \&
.    ds , \&
.    ds ~ ~
.    ds /
.\}
.if t \{\
.    ds ' \\k:\h'-(\\n(.wu*8/10-\*(#H)'\'\h"|\\n:u"
.    ds ` \\k:\h'-(\\n(.wu*8/10-\*(#H)'\`\h'|\\n:u'
.    ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'^\h'|\\n:u'
.    ds , \\k:\h'-(\\n(.wu*8/10)',\h'|\\n:u'
.    ds ~ \\k:\h'-(\\n(.wu-\*(#H-.1m)'~\h'|\\n:u'
.    ds / \\k:\h'-(\\n(.wu*8/10-\*(#H)'\z\(sl\h'|\\n:u'
.\}
.    \" troff and (daisy-wheel) nroff accents
.ds : \\k:\h'-(\\n(.wu*8/10-\*(#H+.1m+\*(#F)'\v'-\*(#V'\z.\h'.2m+\*(#F'.\h'|\\n:u'\v'\*(#V'
.ds 8 \h'\*(#H'\(*b\h'-\*(#H'
.ds o \\k:\h'-(\\n(.wu+\w'\(de'u-\*(#H)/2u'\v'-.3n'\*(#[\z\(de\v'.3n'\h'|\\n:u'\*(#]
.ds d- \h'\*(#H'\(pd\h'-\w'~'u'\v'-.25m'\f2\(hy\fP\v'.25m'\h'-\*(#H'
.ds D- D\\k:\h'-\w'D'u'\v'-.11m'\z\(hy\v'.11m'\h'|\\n:u'
.ds th \*(#[\v'.3m'\s+1I\s-1\v'-.3m'\h'-(\w'I'u*2/3)'\s-1o\s+1\*(#]
.ds Th \*(#[\s+2I\s-2\h'-\w'I'u*3/5'\v'-.3m'o\v'.3m'\*(#]
.ds ae a\h'-(\w'a'u*4/10)'e
.ds Ae A\h'-(\w'A'u*4/10)'E
.    \" corrections for vroff
.if v .ds ~ \\k:\h'-(\\n(.wu*9/10-\*(#H)'\s-2\u~\d\s+2\h'|\\n:u'
.if v .ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'\v'-.4m'^\v'.4m'\h'|\\n:u'
.    \" for low resolution devices (crt and lpr)
.if \n(.H>23 .if \n(.V>19 \
\{\
.    ds : e
.    ds 8 ss
.    ds o a
.    ds d- d\h'-1'\(ga
.    ds D- D\h'-1'\(hy
.    ds th \o'bp'
.    ds Th \o'LP'
.    ds ae ae
.    ds Ae AE
.\}
.rm #[ #] #H #V #F C
.\" ========================================================================
.\"
.IX Title "MONIT 1"
.TH MONIT 1 "www.tildeslash.com" "March 24. 2004" "User Commands"
.SH "NAME"
monit \- utility for monitoring services on a Unix system
.SH "SYNOPSIS"
.IX Header "SYNOPSIS"
\&\fBmonit\fR [options] {arguments}
.SH "DESCRIPTION"
.IX Header "DESCRIPTION"
\&\fBmonit\fR is a utility for managing and monitoring processes,
files, directories and devices on a Unix system. Monit conducts
automatic maintenance and repair and can execute meaningful
causal actions in error situations. E.g. monit can start a
process if it does not run, restart a process if it does not
respond and stop a process if it uses to much resources. You may
use monit to monitor files, directories and devices for changes,
such as timestamps changes, checksum changes or size changes.
.PP
Monit is controlled via an easy to configure control file based
on a free\-format, token-oriented syntax. Monit logs to syslog or
to its own log file and notifies you about error conditions via
customizable alert messages. Monit can perform various \s-1TCP/IP\s0
network checks, protocol checks and can utilize \s-1SSL\s0 for such
checks. Monit provides a http(s) interface and you may use a
browser to access the monit program.
.SH "GENERAL OPERATION"
.IX Header "GENERAL OPERATION"
The behavior of monit is controlled by command-line options
\&\fIand\fR a run control file, \fI~/.monitrc\fR, the syntax of which we
describe in a later section. Command-line options override
\&\fI.monitrc\fR declarations.
.PP
The following options are recognized by monit. However, it is
recommended that you set options (when applicable) directly in
the \fI.monitrc\fR control file.
.Sh "General Options and Arguments"
.IX Subsection "General Options and Arguments"
\&\fB\-c\fR \fIfile\fR
   Use this control file
.PP
\&\fB\-d\fR \fIn\fR
   Run as a daemon once per \fIn\fR seconds
.PP
\&\fB\-g\fR 
   Set group name for start, stop, restart and status
.PP
\&\fB\-l\fR \fIlogfile\fR
   Print log information to this file 
.PP
\&\fB\-p\fR \fIpidfile\fR
   Use this lock file in daemon mode
.PP
\&\fB\-s\fR \fIstatefile\fR
   Write state information to this file
.PP
\&\fB\-I\fR
   Do not run in background (needed for run from init)
.PP
\&\fB\-t\fR
   Run syntax check for the control file
.PP
\&\fB\-v\fR
   Verbose mode, work noisy (diagnostic output)
.PP
\&\fB\-V\fR
   Print version number and patch level
.PP
\&\fB\-h\fR
   Print a help text
.PP
In addition to the options above, monit can be started with one
of the following action arguments; monit will then execute the
action and exit without transforming itself to a daemon.
.PP
\&\fBstart all\fR
   Start all services listed in the control file and 
   enable monitoring for them. If the group option is 
   set, only start and enable monitoring of services in
   the named group.
.PP
\&\fBstart name\fR
   Start the named service and enable monitoring for 
   it. The name is a service entry name from the 
   monitrc file.
.PP
\&\fBstop all\fR     
   Stop all services listed in the control file and 
   disable their monitoring. If the group option is 
   set, only stop and disable monitoring of the services 
   in the named group.
.PP
\&\fBstop name\fR
   Stop the named service and disable its monitoring. 
   The name is a service entry name from the monitrc 
   file. 
.PP
\&\fBrestart all\fR
   Stop and start \fIall\fR services. If the group option 
   is set, only restart the services in the named group.
.PP
\&\fBrestart name\fR
   Restart the named service. The name is a service entry 
   name from the monitrc file.
.PP
\&\fBmonitor all\fR     
   Enable monitoring of all services listed in the
   control file. If the group option is set, only start
   monitoring of services in the named group.
.PP
\&\fBmonitor name\fR
   Enable monitoring of the named service.  The name is
   a service entry name from the monitrc file. Monit will
   also enable monitoring of all services this service 
   depends on.
.PP
\&\fBunmonitor all\fR     
   Disable monitoring of all services listed in the
   control file. If the group option is set, only disable
   monitoring of services in the named group.
.PP
\&\fBunmonitor name\fR
   Disable monitoring of the named service. The name is
   a service entry name from the monitrc file. Monit 
   will also disable monitoring of all services that 
   depends on this service.
.PP
\&\fBstatus\fR
    Print status information for each service.
.PP
\&\fBreload\fR
    Reinitialize a running monit daemon, the daemon will
    reread its configuration, close and reopen log files.
.PP
\&\fBquit\fR
    Kill a monit daemon process
.PP
\&\fBvalidate\fR
   Check all services listed in the control file. This 
   action is also the default behavior when monit runs 
   in daemon mode.
.SH "WHAT TO MONITOR"
.IX Header "WHAT TO MONITOR"
You may use monit to monitor daemon processes or similar programs
running on localhost. Monit is particular useful for monitoring
daemon processes, such as those started at system boot time from
/etc/init.d/. For instance sendmail, sshd, apache and mysql. In
difference to many monitoring systems, monit can act if an error
situation should occur, e.g.; if sendmail is not running, monit
can start sendmail or if apache is using to much system resources
(e.g. if a DoS attack is in progress) monit can stop or restart
apache and send you an alert message. Monit does also monitor
process characteristics, such as; if a process has become a
zombie and how much memory or cpu cycles a process is using.
.PP
You may also use monit to monitor files, directories and devices
on localhost. Monit can monitor these items for changes, such as
timestamps changes, checksum changes or size changes. This is
also useful for security reasons \- you can monitor the md5
checksum of files that should not change.
.PP
You may even use monit to monitor remote hosts. First and
foremost monit is a utility for monitoring and mending services
on localhost, but if a service depends on a remote service,
e.g. a database server or an application server, it might by
useful to be able to test a remote host as well.
.SH "HOW TO MONITOR"
.IX Header "HOW TO MONITOR"
monit is configured and controlled via a control file called
\&\fBmonitrc\fR. The default location for this file is ~/.monitrc. If
this file does not exist, monit will try /etc/monitrc and finally
\&./monitrc.
.PP
A monit control file consists of a series of service entries and
global option statements in a free\-format, token-oriented syntax.
Comments begin with a # and extend through the end of the line.
There are three kinds of tokens in the control file: grammar
keywords, numbers and strings.
.PP
On a semantic level, the control file consists of three types of
statements:
.IP "1. Global set-statements" 4
.IX Item "1. Global set-statements"
A global set-statement starts with the keyword \fIset\fR and the
item to configure.
.IP "2. Global include-statement" 4
.IX Item "2. Global include-statement"
The include statement consists of the keyword \fIinclude\fR and
a glob string.
.IP "3. One or more service entry statements." 4
.IX Item "3. One or more service entry statements."
A service entry starts with the keyword \fIcheck\fR followed by the
service type. 
.PP
This is the hello galaxy version of a monit control file:
.PP
.Vb 3
\& #
\& # monit control file
\& #
.Ve
.PP
.Vb 6
\& set daemon 120 # Poll at 2-minute intervals
\& set logfile syslog facility log_daemon
\& set alert foo@bar.baz
\& set httpd port 2812 and use address localhost
\&     allow localhost   # Allow localhost to connect
\&     allow admin:monit # Allow Basic Auth
.Ve
.PP
.Vb 11
\& check process apache 
\&    with pidfile "/usr/local/apache/logs/httpd.pid"
\&    start program = "/etc/init.d/httpd start"
\&    stop program = "/etc/init.d/httpd stop"
\&    if 2 restarts within 3 cycles then timeout
\&    if totalmem > 100 Mb then alert
\&    if children > 255 for 5 cycles then stop
\&    if cpu usage > 95% for 3 cycles then restart
\&    if failed port 80 protocol http then restart
\&    group server
\&    depends on httpd.conf, httpd.bin
.Ve
.PP
.Vb 5
\& check file httpd.conf 
\&     with path /usr/local/apache/conf/httpd.conf
\&     # Restart apache if the httpd.conf file was changed
\&     if failed checksum 
\&        then exec "/usr/local/apache/bin/apachectl graceful"
.Ve
.PP
.Vb 4
\& check file httpd.bin 
\&     with path /usr/local/apache/bin/httpd
\&     # Alert and unmonitor apache if the binary was changed
\&     if failed checksum then unmonitor
.Ve
.PP
.Vb 2
\& include /etc/monit/mysql.monitrc
\& include /etc/monit/mail/*.monitrc
.Ve
.PP
This example illustrate a service entry for monitoring the apache
web server process as well as related files. The meaning of the
various statements will be explained in the following sections.
.SH "LOGGING"
.IX Header "LOGGING"
monit will log status and error messages to a log file. If
\&\fBsyslog\fR is given as a value for the \fI\-l\fR command-line switch
(or the keyword \fIset logfile syslog\fR is found in the control
file) monit will use the \fBsyslog\fR system daemon to log
messages. To turn off logging, simply do not set the logfile in
the control file (and of course, do not use the \-l switch)
.SH "DAEMON MODE"
.IX Header "DAEMON MODE"
The \fI\-d interval\fR command-line switch runs monit in daemon
mode. You must specify a numeric argument which is a polling
interval in seconds.
.PP
In daemon mode, monit detaches from the console, puts itself in
the background and runs continuously, monitoring each specified
service and then goes to sleep for the given poll interval.
.PP
.Vb 1
\&       Simply invoking
.Ve
.PP
.Vb 1
\&              monit -d 300
.Ve
.PP
will poll all services described in your \fI~/.monitrc\fR file every
5 minutes.
.PP
It is possible and also strongly recommended to set the poll
interval in your ~/.monitrc file by using 'set daemon \fBn\fR',
where \fBn\fR is an integer number of seconds.  If you do this,
monit will always start in daemon mode (as long as no action
arguments are given).
.PP
Monit makes a per-instance lockfile in daemon mode. If you need
more monit instances, you will need more configuration files,
each pointing to its own lockfile.
.PP
Calling monit with a monit daemon running in the background sends
a wakeup signal to the daemon, forcing it to check services
immediately.
.PP
The \fIquit\fR argument will kill a running daemon process instead
of waking it up.
.SH "INIT SUPPORT"
.IX Header "INIT SUPPORT"
Monit can run and be controlled from \fIinit\fR. If monit should
crash, \fIinit\fR will re-spawn a new monit process. Using init to
start monit is probably the best way to run monit if you want to
be certain that you always have a running monit daemon on your
system. (It's obvious, but never the less worth to stress; Make
sure that the control file does not have any syntax errors before
you start monit from init. Also, make sure that if you run monit
from init, that you do not start monit from a startup scripts as
well).
.PP
To setup monit to run from init, you can either use the 'set
init' statement in monit's control file or use the \-I option from
the command line. Here is a sample /etc/inittab entry for monit:
.PP
.Vb 2
\&  # Run monit in standard run-levels
\&  mo:2345:respawn:/usr/local/bin/monit -Ic /etc/monitrc
.Ve
.PP
After you have modified init's configuration file, you can run
the following command to re-examine /etc/inittab and start monit:
.PP
.Vb 1
\&  telinit q
.Ve
.PP
For systems without telinit:
.PP
.Vb 1
\&  kill -1 1
.Ve
.PP
If monit is used to monitor services that are also started at
boot time (e.g. services started via \s-1SYSV\s0 init rc scripts or via
inittab) then in some situations a race condition can occur. That
is; if a service is slow to start, monit can assume that the
service is not running and possibly try to start it and raise an
alert, while, in fact the service is already about to start or
already in its startup sequence. Please see the \s-1FAQ\s0 for solutions
to this problem.
.SH "INCLUDE FILES"
.IX Header "INCLUDE FILES"
The monit control file, \fImonitrc\fR, can include additional files.
This feature helps to maintain a certain structure or to place
repeating settings into one file. Include statements can be
placed at virtually any spot. The syntax is the following:
.PP
.Vb 1
\&  INCLUDE globstring
.Ve
.PP
The globstring is any kind of string as defined in \fIglob\fR\|(7).
Thus, you can refer to a single file or you can load several
files at once.  In case you want to use whitespace in your string
the globstring need to be embedded into quotes (') or double
quotes ("). For example,
.PP
.Vb 1
\&  INCLUDE "/etc/monit/monit configuration files/printer.*.monitrc"
.Ve
.PP
loads any file matching the single globstring.  If the globstring
matches a directory instead of a file, it is silently ignored.
.PP
\&\fI\s-1INCLUDE\s0\fR statements in included files are parsed as in the main
control file.  
.PP
If the globstring matches several results, the files are included
in a non sorted manner.  If you need to rely on a certain order,
you might need to use single \fIinclude\fR statements.
.SH "GROUP SUPPORT"
.IX Header "GROUP SUPPORT"
Service entries in the control file, \fImonitrc\fR, can be grouped
together by the \fIgroup\fR statement. The syntax is simply (keyword
in capital):
.PP
.Vb 1
\&  GROUP groupname
.Ve
.PP
With this statement it is possible to group similar service
entries together and manage them as a whole. Monit provides
functions to start, stop and restart a group of services, like
so:
.PP
To start a group of services from the console:
.PP
.Vb 1
\&  monit -g <groupname> start
.Ve
.PP
To stop a group of services:
.PP
.Vb 1
\&  monit -g <groupname> stop
.Ve
.PP
To restart a group of services:
.PP
.Vb 1
\&  monit -g <groupname> restart
.Ve
.SH "MONITORING MODE"
.IX Header "MONITORING MODE"
Monit supports three monitoring modes per service: \fIactive\fR,
\&\fIpassive\fR and \fImanual\fR. See also the example section below for
usage of the mode statement.
.PP
In \fIactive\fR mode, monit will monitor a service and in case of
problems monit will act and raise alerts, start, stop or restart
the service. Active mode is the default mode.
.PP
In \fIpassive\fR mode, monit will passively monitor a service and
specifically \fBnot\fR try to fix a problem, but it will still raise
alerts in case of a problem.
.PP
For use in clustered environments there is also a \fImanual\fR
mode. In this mode, monit will enter \fIactive\fR mode \fBonly\fR if a
service was brought under monit's control, for example by
executing the following command in the console:
.PP
.Vb 2
\&  monit start sybase 
\&  (monit will call sybase's start method and enable monitoring)
.Ve
.PP
If a service was not started by monit or was stopped or disabled
for example by:
.PP
.Vb 2
\&  monit stop sybase 
\&  (monit will call sybase's stop method and disable monitoring)
.Ve
.PP
monit will not monitor the service. This allows for having
services configured in monitrc and start it with monit only if it
should run. This feature can be used to build a simple failsafe
cluster. To see how, read more about how to setup a cluster with
monit using the \fIheartbeat\fR system in the examples sections
below.
.SH "ALERT MESSAGES"
.IX Header "ALERT MESSAGES"
monit will raise an email alert if:
.PP
.Vb 12
\& o A service timed out
\& o A service was stopped
\& o A port connection failed
\& o A resource statement match
\& o A permission error occurred
\& o A file checksum was changed
\& o A timestamp test didn't pass
\& o A file size test didn't pass
\& o A service was started or restarted
\& o A file/directory/device uid error occurred
\& o A file/directory/device gid error occurred
\& o Monitoring of a service was disabled (unmonitor)
.Ve
.PP
You should use an alert statement to notify monit that you want
alert messages sent to an email address. If you do not specify an
alert statement, monit will not send alert messages.
.PP
There are two forms of alert statement:
.PP
.Vb 2
\& o Global - common for all services
\& o Local  - per service
.Ve
.PP
In both cases you can use more than one alert statement. In other
words, you can send many different emails to many different
addresses. (Just to make it clear, in case you now got a new
business idea; monit is not suitable for sending spam).
.PP
Recipients in the global and in the local lists are alerted when
an error event occurs. If the same email address is in the global
and in the local list, monit will still only send one alert.
Local (per service) defined alert email addresses override global
addresses in case of a conflict. Finally, you may choose to only
use a global alert list, a local per service list or both.
.PP
The full syntax for the alert statement is as follows (keywords
are in capital):
.Sh "Setting a global alert statement"
.IX Subsection "Setting a global alert statement"
In case of an event on any monitored service monit will send an
alert to recipients in the global list which have registred
interest for the event type.
.IP "\s-1SET\s0 \s-1ALERT\s0 mail-address [{events}] [\s-1MAIL\-FORMAT\s0 {mail\-format}]" 4
.IX Item "SET ALERT mail-address [{events}] [MAIL-FORMAT {mail-format}]"
.PP
Simply using the following in the global section of monitrc:
.PP
.Vb 1
\& set alert foo@bar
.Ve
.PP
will send a default email to the address foo@bar whenever an
event occurred on any service. Such an event may be that a
service timed out, a service was started or stopped and so on.
.Sh "Setting a local alert statement"
.IX Subsection "Setting a local alert statement"
Each service can have its own recipient list.
.IP "\s-1ALERT\s0 mail-address [{events}] [\s-1MAIL\-FORMAT\s0 {mail\-format}]" 4
.IX Item "ALERT mail-address [{events}] [MAIL-FORMAT {mail-format}]"
.PP
If you only want an alert message sent for certain events for
certain service(s), for example only for timeout events or only
if a service was restarted, then postfix the alert-statement
with a filter block, respectively:
.PP
.Vb 4
\& check process myproc with pidfile /var/run/my.pid
\&  ...
\&  alert foo@bar only on { timeout } and 
\&  alert foo@bar only on { restart }
.Ve
.PP
(\fIonly\fR and \fIon\fR are noise keywords, ignored by monit. As a
side note; Noise keywords are used in the control file grammar to
make an entry resemble English and thus make it easier to read
(or, so goes the philosophy). The full set of available noise
keywords are listed below in the Control File section)
.PP
It is also possible to combine alert events and send mail to
different email addresses depending on the events. Like so:
.PP
.Vb 3
\& alert foo@bar { restart, timeout, resource, connection }
\& alert security@bar on { checksum, permission, uid, gid, stop }
\& alert manager@bar
.Ve
.PP
This will send an alert message to foo@bar whenever a restart,
timeout, resource or connection error occurs and a message to
security@bar if a checksum, permission, uid, gid or stop error
occurs.  And finally, a message to manager@bar whenever any error
event occurs.
.PP
The following alert\-statement:
.PP
.Vb 13
\& alert foo@bar {uid
\&                gid
\&                size
\&                stop
\&                start
\&                restart
\&                timeout
\&                resource
\&                checksum
\&                timestamp
\&                unmonitor
\&                connection
\&                permission}
.Ve
.PP
is equivalent to:
.PP
.Vb 1
\& alert foo@bar
.Ve
.PP
The following example will alert foo@bar on all events on all
services by default, except the service mybar which will send an
alert only on timeout. The trick is based on the fact that local
definition of the same recipient overrides the global setting
(including registered events and mail format):
.PP
.Vb 1
\& set alert foo@bar
.Ve
.PP
.Vb 1
\& check process myfoo with pidfile /var/run/myfoo.pid
.Ve
.PP
.Vb 2
\& check process mybar with pidfile /var/run/mybar.pid
\&  alert foo@bar only on { timeout }
.Ve
.Sh "Alert message layout"
.IX Subsection "Alert message layout"
monit provides a default mail message layout that is short and to
the point. Here's an example of a standard alert mail sent by
monit:
.PP
.Vb 4
\& From: monit@tildeslash.com
\& Subject: monit alert -- Restarted apache 
\& To: hauk@tildeslash.com
\& Date: Thu, 04 Sep 2003 02:33:03 +0200
.Ve
.PP
.Vb 1
\& Restarted Service apache
.Ve
.PP
.Vb 2
\&        Date: Thu, 04 Sep 2003 02:33:03 +0200
\&        Host: www.tildeslash.com
.Ve
.PP
.Vb 2
\& Your faithful employee,
\& monit
.Ve
.PP
If you want to, you can change the format of this message with
the optional \fImail-format\fR statement. The syntax for this
statement is as follows:
.PP
.Vb 7
\& mail-format {
\&      from: monit@localhost
\&   subject: $SERVICE $EVENT at $DATE
\&   message: Monit restarted $SERVICE at $DATE on $HOST. 
\&            Yours sincerely,
\&            monit
\& }
.Ve
.PP
Where the keyword \fIfrom:\fR is the email address monit should
pretend it is sending from. It does not have to be a real mail
address, but it must be a proper formated mail address, on the
form: name@domain. The keyword \fIsubject:\fR is for the email
subject line. The subject must be on only \fIone\fR line. The
\&\fImessage:\fR keyword denotes the mail body. If used, this keyword
should always be the last in a mail-format statement.  The mail
body can be as long as you want and must \fBnot\fR contain the '}'
character.
.PP
All of these format keywords are optional but you must provide at
least one. Thus if you only want to change the from address monit
is using you can do:
.PP
.Vb 1
\& alert foo@bar with mail-format { from: bofh@xyzzy.no }
.Ve
.PP
From the previous example you will notice that 4 special
variables was used. If used, they will be substituted and
expanded into the text with these values:
.IP "\(bu" 4
\&\fI$EVENT\fR A string describing the event that occurred. The values
are fixed and are, \*(L"Started\*(R", \*(L"Stopped\*(R", \*(L"Restarted\*(R", Checksum
changed\*(L", \*(R"Resource limit matched\*(L", \*(R"Timeout\*(L", \*(R"Size rule
matched\*(L", \*(R"Permission error\*(L", \*(R"Connection test failed\*(L", \*(R"\s-1UID\s0
error\*(L", \*(R"\s-1GID\s0 error\*(L", \*(R"Monitoring disabled\*(L" and \*(R"Timestamp rule
matched".
.IP "\(bu" 4
\&\fI$SERVICE\fR The service entry name in monitrc
.IP "\(bu" 4
\&\fI$DATE\fR The current time and date (\s-1RFC\s0 822 date style).
.IP "\(bu" 4
\&\fI$HOST\fR The name of the host monit is running on
.Sh "Setting a global mail format"
.IX Subsection "Setting a global mail format"
Finally, it is possible to set a standard mail format with the
following global set-statement (keywords are in capital):
.IP "\s-1SET\s0 MAIL-FORMAT {mail\-format}" 4
.IX Item "SET MAIL-FORMAT {mail-format}"
.PP
Format set with this statement will apply to every alert
statement that does \fInot\fR have its own specified mail\-format.
This statement is most useful for setting a default from address
for messages sent by monit, like so:
.PP
.Vb 1
\& set mail-format { from: monit@foo.bar.no }
.Ve
.Sh "Setting a mail server for alert messages"
.IX Subsection "Setting a mail server for alert messages"
The mail server monit should use to send alert messages is
defined with a global set statement (keywords are in capital and
optional statements in [brackets]):
.PP
.Vb 1
\& SET MAILSERVER {host name|ip-address}+ [with TIMEOUT X SECONDS]
.Ve
.PP
As you can see, it is possible to set several \s-1SMTP\s0 servers and if
monit cannot connect to the first server in the list it will try
the second server and so on. Monit has a default 5 seconds
connection timeout and if the \s-1SMTP\s0 server is slow, monit could
timeout when connecting or reading from the server.  You can use
the optional timeout statement to explicit set the timeout to a
higher value if needed. Here is an example for setting several
mail servers:
.PP
.Vb 2
\& set mailserver mail.tildeslash.com, mail.foo.bar, localhost 
\&     with timeout 15 seconds
.Ve
.PP
Here monit will first try to connect to the server
\&\*(L"mail.tildeslash.com\*(R", if this server is down monit will try
\&\*(L"mail.foo.bar\*(R" and finally \*(L"localhost\*(R". We do also set an
explicit connect and read timeout; If monit cannot connect to the
first \s-1SMTP\s0 server in the list within 15 seconds it will try the
next server and so on. The \fIset mailserver ..\fR statement is
optional and if not defined monit defaults to use localhost as
the \s-1SMTP\s0 server.
.SH "SERVICE TIMEOUT"
.IX Header "SERVICE TIMEOUT"
\&\fBmonit\fR provides a service timeout mechanism for situations
where a service simply refuses to start or respond over a longer
period. In cases like this, and particularly if monit's poll-cycle
is low, monit will simply increase the machine load by trying to
restart the service.
.PP
The timeout mechanism monit provides is based on two variables,
i.e. the number the service has been started and the number of
poll\-cycles. For example, if a service had \fIx\fR restarts within
\&\fIy\fR poll-cycles (where \fIx\fR <= \fIy\fR) then monit will timeout and
not (re)start the service on the next cycle. If a timeout occurs
monit will send you an alert message if you have register
interest for this event. 
.PP
The syntax for the timestamp statement is as follows (keywords
are in capital):
.IP "\s-1IF\s0 \s-1NUMBER\s0 \s-1RESTART\s0 \s-1NUMBER\s0 \s-1CYCLE\s0(S) \s-1THEN\s0 \s-1TIMEOUT\s0" 4
.IX Item "IF NUMBER RESTART NUMBER CYCLE(S) THEN TIMEOUT"
.PP
Where the first number is the number of service restarts and the
second, the number of poll\-cycles. If the number of cycles was
reached without a timeout, the service start-counter is reset to
zero. This provides some granularity to catch exceptional cases
and do a service timeout, but let occasional service start and
restarts happen without having an accumulated timeout.
.PP
Here is an example where monit will timeout (not check the
service) if the service was restarted 2 times within 3 cycles:
.PP
.Vb 1
\& if 2 restarts within 3 cycles then timeout
.Ve
.PP
To have monit check the service again after a timeout, run 'monit
monitor service' from the command line. This will remove the
timeout lock in the daemon and make the daemon start and check
the service again.
.SH "SERVICE TESTS"
.IX Header "SERVICE TESTS"
monit provides several if-tests you may utilize in a service
entry to test a service. Each if-test statement is on the
following general format:
.IP "\s-1IF\s0 <\s-1TEST\s0> \s-1THEN\s0 \s-1ACTION\s0" 4
.IX Item "IF <TEST> THEN ACTION"
.PP
If the <\s-1TEST\s0> should validate to true then the selected action is
executed. You must select an action to be executed from this
list:
.IP "\(bu" 4
\&\fB\s-1ALERT\s0\fR sends the user an alert event.
.IP "\(bu" 4
\&\fB\s-1RESTART\s0\fR restarts the service \fIand\fR sends an alert. Restart is
conducted by first calling the service's registered stop method
and then the service's start method.
.IP "\(bu" 4
\&\fB\s-1STOP\s0\fR stops the service by calling the service's registered stop
method \fIand\fR send an alert. If monit stops a service it will not
be checked by monit anymore nor restarted again later. To
reactivate monitoring of the service again you must explicitly
enable monitoring from the web interface or from the console,
e.g. 'monit monitor apache'.
.IP "\(bu" 4
\&\fB\s-1EXEC\s0\fR may be used to execute an arbitrary program \fIand\fR send
an alert. If you choose this action you must state the program to
be executed and if the program require arguments you must enclose
the program and its arguments in a quoted string. You may
optionally specify the uid and gid the executed program should
switch to upon start. For instance:
.Sp
.Vb 2
\& exec "/usr/local/tomcat/bin/startup.sh"
\&      as uid nobody and gid nobody
.Ve
.Sp
This may be useful if the program to be started cannot change to
a lesser privileged user and group. This is typically needed for
Java Servers. Remember, if monit is run by the superuser, then
all programs executed by monit will be started with superuser
privileges unless the uid and gid extension was used.
.IP "\(bu" 4
\&\fB\s-1UNMONITOR\s0\fR will disable monitoring of the service \fIand\fR send
an alert. The service will not be checked by monit anymore nor
restarted again later.  To reactivate monitoring of the service
you must explicitly enable monitoring from monit's web interface
or from the console using the monitor argument.

examples/monit-4.2/contrib/monit.php
<?php
//Script to forward a request on a web server to the Monit webserver.
//Allows the Monit information screen to be viewed remotely, without 
//leaving an additional port open on the server.
//
//Place this script on your webserver.
//Use a password protected area if you like.
//Set the configuration information below to match your monit installation. 
//With this script you are accessing monit from the localhost, so you don't
//need to allow external access through your firewall.
//
//Requires PHP, and a working webserver!
//Copyright 2003 GNU General Public License Version 2
//Author David Fletcher, <david@megapico.co.uk>

//Configuration - set this to match the information in /etc/monitrc
//Set information about the monit server address, port and login details.
$addr = 'localhost';
$port = 2812;
$path = '/';
$user = 'admin';
$pass = 'monit';
$timeout = 30;
//End of configuration

$urlHandle = fsockopen($addr, $port, $errno, $errstr, $timeout);
socket_set_timeout($urlHandle, $timeout);

$urlString = "GET $path HTTP/1.0\r\nHost: $addr\r\nConnection: Keep-Alive\r\nUser-Agent: MonitPHP\r\n";
$urlString .= "Authorization: Basic ".base64_encode("$user:$pass")."\r\n";
$urlString .= "\r\n";
      
fputs($urlHandle, $urlString);
$response = fgets($urlHandle);

$endHeader = false;                           // Strip initial header information
while ( !$endHeader){
  if (fgets($urlHandle) == "\r\n")
    $endHeader = true;
}

$info = '';                                  // Generate a string to send to the screen

while (!feof($urlHandle)) {
  $info.=fgets($urlHandle);
}

fclose ($urlHandle);

echo $info;

?>
